server/src/apub/activities.rs

```diff
@@ -26,48 +26,59 @@ pub fn populate_object_props(
   Ok(())
 }
 
-pub fn send_activity_to_community<A>(
+pub async fn send_activity_to_community<A>(
   creator: &User_,
-  conn: &PgConnection,
   community: &Community,
   to: Vec<String>,
   activity: A,
-) -> Result<(), Error>
+  client: &Client,
+  pool: DbPool,
+) -> Result<(), LemmyError>
 where
-  A: Activity + Base + Serialize + Debug,
+  A: Activity + Base + Serialize + Debug + Clone + Send + 'static,
 {
-  insert_activity(&conn, creator.id, &activity, true)?;
+  insert_activity(creator.id, activity.clone(), true, pool.clone()).await?;
 
   // if this is a local community, we need to do an announce from the community instead
   if community.local {
-    Community::do_announce(activity, &community, creator, conn)?;
+    Community::do_announce(activity, &community, creator, client, pool).await?;
   } else {
-    send_activity(&activity, creator, to)?;
+    send_activity(client, &activity, creator, to).await?;
   }
+
   Ok(())
 }
 
 /// Send an activity to a list of recipients, using the correct headers etc.
-pub fn send_activity<A>(activity: &A, actor: &dyn ActorType, to: Vec<String>) -> Result<(), Error>
+pub async fn send_activity<A>(
+  client: &Client,
+  activity: &A,
+  actor: &dyn ActorType,
+  to: Vec<String>,
+) -> Result<(), LemmyError>
 where
-  A: Serialize + Debug,
+  A: Serialize,
 {
-  let json = serde_json::to_string(&activity)?;
-  debug!("Sending activitypub activity {} to {:?}", json, to);
+  let activity = serde_json::to_string(&activity)?;
+  debug!("Sending activitypub activity {} to {:?}", activity, to);
+
   for t in to {
     let to_url = Url::parse(&t)?;
     if !is_apub_id_valid(&to_url) {
       debug!("Not sending activity to {} (invalid or blocklisted)", t);
       continue;
     }
-    let request = Request::post(t).header("Host", to_url.domain().unwrap());
-    let signature = sign(&request, actor)?;
-    let res = request
-      .header("Signature", signature)
```

Comment:
What's up with the signature going away here?

Reply:
it's handled in the sign function